src/hooks/useAuthentication.ts

```diff
@@ -1,14 +1,14 @@
-import { useSigninMutation } from "../app/services/AuthService";
+import { useSelector } from "react-redux";
+import { store } from "../app/store/store";
 
 const useAuthentication = () => {
-  const [, { data }] = useSigninMutation({ fixedCacheKey: "shared-auth" });
-
-  if (!data) return { isAuthenticated: false };
+  const auth = useSelector((state: ReturnType<typeof store.getState>) => state.auth);
+  if(!auth.user) return { isAuthenticated: false };
 
   return {
-    isAuthenticated: data.accessToken != null,
-    userId: data.userId,
-    accessToken: data.accessToken,
+    isAuthenticated: auth.user.accessToken != null,
+    userId: auth.user.userId,
+    accessToken: auth.user.accessToken,
   };
```

Comment:
Acá si hacemos que en el useSelector en vez de devolver `state.auth` devuelva el `state.auth.user` nos ahorramos después tenes que acceder al user en las otras lineas


```suggestion
  const user = useSelector((state: ReturnType<typeof store.getState>) => state.auth.user);
  if(!user) return { isAuthenticated: false };

  return {
    isAuthenticated: user.accessToken != null,
    userId: user.userId,
    accessToken: user.accessToken,
  };
};
```